bounding-boxes/src/coco_to_datasaur_schemas.py

```diff
@@ -0,0 +1,264 @@
+    )
+    images = coco_json["images"]
+
+    retval: List[dict] = []
+    for image in images:
+        annotations_by_images = [
+            annot
+            for annot in coco_json["annotations"]
+            if annot["image_id"] == image["id"]
+        ]
+
+        bbox_labels = [
+            bbox_label_from_coco_annotation(annot, bbox_label_set)
+            for annot in annotations_by_images
+        ]
+
+        schema = DatasaurSchema(
+            version="1",
+            data=DSBBoxProjectData(
+                kinds=["BBOX_BASED"],
+                bboxLabels=bbox_labels,
+                bboxLabelSets=[bbox_label_set],
+                document=GenericIdAndName(name=image["file_name"], id=None),
+                project=None,
+                pages=[
+                    DSPage(
+                        pageIndex=0,
+                        pageHeight=floor(image["height"]),
+                        pageWidth=floor(image["width"]),
+                    )
+                ],
+            ),
+        )
+
+        retval.append(asdict(schema))
+
+    return retval
+
+
+def main() -> None:
+    parser = ArgumentParser(prog="coco_to_datasaur_schemas")
+    parser.add_argument("coco_filepath", type=str, help="Path to COCO JSON file")
+    parser.add_argument(
+        "--custom_labelset",
+        type=str,
+        help="Path to custom labelset JSON file (useful for specifying DROPDOWN attributes)",
+    )
+    parser.add_argument(
+        "--outdir",
+        type=str,
+        help="Output directory for Datasaur schemas",
+        default="./outdir/",
+    )
+    parser.add_argument("--log-level", type=str, default="INFO")
+    args = parser.parse_args()
+    logging.basicConfig(level=args.log_level, format="%(message)s")
+
+    coco_filepath = os.path.abspath(args.coco_filepath)
+    log("reading COCO JSON file", filepath=coco_filepath)
+    with open(coco_filepath) as f:
+        json_data = json.load(f)
+
+    custom_labelset = None
+    if args.custom_labelset:
+        custom_labelset_filepath = os.path.abspath(args.custom_labelset)
+        with open(custom_labelset_filepath) as f:
+            custom_labelset = json.load(f)
+
+    log("converting COCO to Datasaur Schema")
+    schemas = coco_to_datasaur_schemas(json_data, custom_labelset)
+
+    outdir = os.path.abspath(args.outdir)
+    os.makedirs(outdir, exist_ok=True)
+    log("writing to file", directory=outdir)
+    for schema in schemas:
+        filename = schema["data"]["document"]["name"].split(".")[0] + ".json"
+
+        with open(os.path.join(outdir, filename), "w") as f:
+            json.dump(scrub(schema), f, indent=2)
+
+
+def bbox_label_classes_from_coco(
+    coco_categories: List[dict],
+    custom_labelset: Any | None,
+) -> List[DSBBoxLabelClass]:
+    def defaults(obj: dict | None, key: str, default: Any):
+        return obj[key] if obj and key in obj else default
```

Comment:
dikeluarin aja? bisa masuk common mungkin?

Reply:
lifted to commons 👍🏼